tests/test_interrupted_sync.py

```diff
@@ -79,24 +94,27 @@ def test_run(self):
 
         # hardcoding the updated state to ensure atleast 1 record in resuming (2nd) sync.
         # values have been provided after reviewing the max bookmark value for each of the streams
-        new_state = {'bookmarks':
-                     {'currently_sync_stream': 'customers',
-                      'collects': {'updated_at': '2023-01-01T09:08:28.000000Z'},
+        currently_syncing_stream = random.choice(list(expected_streams))
+        LOGGER.info(f"Randomly selected currently syncing stream: {currently_syncing_stream}")
+
+        stream_groups = self.group_streams(first_sync_order, currently_syncing_stream)
+        completed_streams = stream_groups.get('completed')
+        yet_to_be_synced_streams = stream_groups.get('yet_to_be_synced')
+
+        base_state = {'bookmarks':
+                     {'currently_sync_stream': currently_syncing_stream,
                       'customers': {'updated_at': '2023-03-28T18:53:28.000000Z'},
                       'events': {'created_at': '2023-01-22T05:05:53.000000Z'},
-                      #'metafields': {'updated_at': '2023-01-07T21:18:05.000000Z'},
-                      #'orders': {'updated_at': '2023-01-20T05:09:01.000000Z'},
-                      #'products': {'updated_at': '2023-01-20T05:10:05.000000Z'},
+                      'metafields': {'updated_at': '2023-01-07T21:18:05.000000Z'},
+                      'orders': {'updated_at': '2023-01-22T05:07:44.000000Z'},
+                      'products': {'updated_at': '2023-01-22T05:05:56.000000Z'},
                       'transactions': {'created_at': '2022-06-26T00:06:38-04:00'}
                       }}
 
-        currently_syncing_stream = new_state['bookmarks']['currently_sync_stream']
-        completed_streams = {'collects', 'customers', 'events', 'transactions'}
-        completed_streams.remove(currently_syncing_stream)
-        yet_to_be_synced_streams = {'metafields', 'orders', 'products'}
+        new_state = {'bookmarks': {key: val for (key, val) in base_state['bookmarks'].items()
+            if key not in yet_to_be_synced_streams}}
```

Comment:
I think we should have a style change here to help the meaning of what is happening shine through.

```suggestion
        new_state = {
            'bookmarks': {
                key: val
                for key, val in base_state['bookmarks'].items()
                if key not in yet_to_be_synced_streams
            }
        }
```

Reply:
Updated.